examples/src/test/java/jdocs/kafka/KafkaDocExample.java

```diff
@@ -4,9 +4,213 @@
 
 package jdocs.kafka;
 
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Supplier;
+
+import akka.Done;
+import akka.NotUsed;
+import akka.actor.typed.ActorSystem;
+import akka.actor.typed.javadsl.Behaviors;
+import akka.kafka.ProducerSettings;
+import akka.kafka.javadsl.SendProducer;
+import akka.projection.Projection;
+import akka.projection.ProjectionId;
+import akka.projection.javadsl.ExactlyOnceProjection;
+import akka.projection.javadsl.SourceProvider;
+import akka.projection.jdbc.javadsl.JdbcHandler;
+import akka.projection.jdbc.javadsl.JdbcProjection;
+import akka.projection.kafka.GroupOffsets;
+import akka.projection.kafka.javadsl.KafkaSourceProvider;
+import akka.stream.javadsl.Source;
+import jdocs.jpa.HibernateSessionProvider;
+import org.apache.kafka.clients.producer.ProducerRecord;
+import org.apache.kafka.clients.producer.RecordMetadata;
+import org.apache.kafka.common.serialization.StringSerializer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+//#imports
+import akka.kafka.ConsumerSettings;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.serialization.StringDeserializer;
+
+//#imports
+
+import static jdocs.jpa.HibernateSessionProvider.HibernateJdbcSession;
+
 public interface KafkaDocExample {
 
   //#todo
   // TODO
   //#todo
+
+  //#handler
+  public class WordCountHandler extends JdbcHandler<ConsumerRecord<String, String>, HibernateJdbcSession> {
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    private final ProjectionId projectionId;
+
+    private final Map<String, Integer> state = new HashMap<>();
+
+    public WordCountHandler(ProjectionId projectionId) {
+      this.projectionId = projectionId;
+    }
+
+    @Override
+    public void process(HibernateJdbcSession session, ConsumerRecord<String, String> envelope) {
```

Comment:
I guess the goal here is to make it a default handler (ie: `CompletionState<Done>`) and demo at-least-once. Correct?

Reply:
No, consuming should be exactlyOnce. I have created issue https://github.com/akka/akka-projection/issues/256 for making the example more interesting.